include/openenclave/advanced/allocator.h

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+/**
+ * @file allocator.h
+ *
+ * This file defines the interface that pluggable allocators must implement.
+ * See
+ * https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/PluggableAllocators.md
+ *
+ */
+
+#ifndef OE_ADVANCED_ALLOCATOR_H
+#define OE_ADVANCED_ALLOCATOR_H
+
+#include <openenclave/bits/types.h>
+
+/**
+ * @cond IGNORE
+ */
+OE_EXTERNC_BEGIN
+
+/**
+ * @endcond
+ */
+
+/**
+ * Callback for initializing the allocator.
+ *
+ * This function will be called by oecore at an appropriate time during enclave
+ * initialization. The allocator is expected to perform any global
+ * initialization in this function. This function is called prior to any global
+ * initialization.
```

Comment:
Consider clarifying the subject of the global initialization in either case:
``` suggestion 
 * The allocator is expected to perform any global initialization needed by the
 * allocator in this function. This function is called prior to other enclave
 * initialization methods, which can depend on the allocator being ready for
 * use.
```

Reply:
Thanks!